### DISCUSSION TITLE
How does multi LoRA works?

### DISCUSSION THREAD

**Discussion by kamwoh** [huggingface/peft#997] [Q&A]
Does anyone have the reference or the right keywords to understand how multi-LoRA works?

I tried to search online but seems I don't see any research papers discussing it.

**Comment by BenjaminBossan** [comment]
Could you give more details, where did you see "multi LoRA" being mentioned?

**Reply by kamwoh** [reply]
Thanks for the reply, what I meant was that the one in this post demonstrates how to fuse two different LoRA weights together and generate an image with a combined style.

https://huggingface.co/docs/diffusers/main/en/training/lora

I am not sure whether this is a suitable platform to ask or not, I didn't see that github.com/diffusers has a discussion place so I came to here.

**Reply by BenjaminBossan** [reply]
Indeed, this functionality is currently not based on PEFT, though we're working on transitioning diffusers to use PEFT under the hood. A possible place to ask diffusers questions can be on the HF forums:

https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63

Regarding your initial question, I'm not sure if there are any research papers on using multiple LoRAs, this is more something that was developed by the community. If you look at how LoRAs work in general, I think it becomes clear that there is, in theory, no limitation to the amount of LoRA adapters that can be added to a model at once. They are just treated as independent of each other. In practice, this seems to work well, which why support for it was added.
